ledger/participant-integration-api/src/main/scala/platform/metrics/MetricsOwner.scala

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.metrics
+
+import java.util.concurrent.TimeUnit
+
+import com.codahale.metrics.{MetricRegistry, SharedMetricRegistries}
+import com.daml.ledger.resources.{Resource, ResourceContext, ResourceOwner}
+import com.daml.metrics.{JvmMetricSet, Metrics}
+import com.daml.platform.config.MetricsConfig
+import com.daml.platform.config.MetricsConfig.MetricRegistryType
+import io.opentelemetry.api.metrics.Meter
+
+import scala.concurrent.Future
+
+case class MetricsOwner(meter: Meter, config: MetricsConfig, name: String)
+    extends ResourceOwner[Metrics] {
+  override def acquire()(implicit
+      context: ResourceContext
+  ): Resource[Metrics] = {
+    val meterRegistry = config.registryType match {
+      case MetricRegistryType.JvmShared =>
+        SharedMetricRegistries.getOrCreate(name)
+      case MetricRegistryType.New =>
+        new MetricRegistry
+    }
+    val reporter = Option.when(config.enabled) {
+      val runningReporter = config.reporter
+        .register(meterRegistry)
+      runningReporter.start(config.reportingInterval.toMillis, TimeUnit.MILLISECONDS)
+      runningReporter
+    }
+
+    meterRegistry.registerAll(new JvmMetricSet)
```

Comment:
Need to think about how we will get JVM metrics into the opentelemetry to allow a like-for-like experience.

Reply:
Additionally and more importantly by moving this line from sandbox-on-x to this shared class canton will pick up on this.  This is not desirable as canton have a config flag that determines if JVM metrics should be added.  So probably best to add  a helper method that can be:
- built out with support for opentelemtry
- called from sandbox and canton as required.